rs/boundary_node/rate_limits/canister/canister.rs

```diff
@@ -22,6 +22,37 @@ use rate_limits_api::{
 const REGISTRY_CANISTER_ID: &str = "rwlgt-iiaaa-aaaaa-aaaaa-cai";
 const REGISTRY_CANISTER_METHOD: &str = "get_api_boundary_node_ids";
 
+const CANISTER_UPDATE_METHODS: [&str; 2] = ["add_config", "disclose_rules"];
+
+#[inspect_message]
+fn inspect_message() {
+    // In order for this hook to succeed, accept_message() must be invoked.
+    let caller_id: Principal = ic_cdk::api::caller();
+    let called_method = ic_cdk::api::call::method_name();
+
+    // If the called method is not an update method, accept the message.
+    if !CANISTER_UPDATE_METHODS.contains(&called_method.as_str()) {
```

Comment:
This would allow replicated execution of query methods - is that desirable?
In any case, I wouldn't allow calls to non-existing methods.